EXILED/Exiled.Events/Patches/Events/Scp173/Scp173BeingObserved.cs

```diff
@@ -0,0 +1,82 @@
+// -----------------------------------------------------------------------
+// <copyright file="Scp173BeingObserved.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Events.Scp173
+{
+    using System.Collections.Generic;
+    using System.Reflection.Emit;
+
+    using Exiled.API.Features.Pools;
+    using Exiled.Events.Attributes;
+    using Exiled.Events.EventArgs.Scp173;
+    using HarmonyLib;
+    using PlayerRoles.PlayableScps.Scp173;
+    using PlayerRoles.Subroutines;
+    using PluginAPI.Events;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    /// Patches <see cref="Scp173ObserversTracker.IsObservedBy" />.
+    /// Adds the <see cref="Handlers.Scp173.Scp173BeingObserved" /> event.
+    /// </summary>
+    [EventPatch(typeof(Handlers.Scp173), nameof(Handlers.Scp173.Scp173BeingObserved))]
+    [HarmonyPatch(typeof(Scp173ObserversTracker), nameof(Scp173ObserversTracker.IsObservedBy))]
+    internal static class Scp173BeingObserved
+    {
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Pool.Get(instructions);
+
+            Label continueLabel = generator.DefineLabel();
+
+            const int offset = -4;
+            int index = newInstructions.FindIndex(i => i.Is(OpCodes.Call, Field(typeof(EventManager), nameof(EventManager.ExecuteEvent)))) + offset;
```

Comment:
```suggestion
            int index = newInstructions.FindIndex(i => i.Is(OpCodes.Call, Method(typeof(EventManager), nameof(EventManager.ExecuteEvent)))) + offset;
```

Reply:
Also this may be an ambigious reference. Hope that you will test your transpiler

Reply:
Zer0Two told me to do that xd

Reply:
And I couldn't really test it sadly